### DISCUSSION TITLE
Metrics

### DISCUSSION THREAD

**Discussion by gramian** [ArcadeData/arcadedb#551] [Q&A] [answered]
Dear Community,

OrientDB has some means to collect metrics (see [Profiler](https://orientdb.com/docs/last/ee/Profiler.html)) on databases, the server, and a cluster; including HTTP endpoints to obtain these metrics. Are facilities for metrics in ArcadeDB already available (and I overlooked it), is this a planned feature, or maybe unnecessary as similarly useful metrics could be collected from outside of the server by the system?

Thank you for your input

**Comment by lvca** [comment, marked as answer]
We have our own profiler. You can automatically dump the metrics every X seconds by setting this at startup:

`java ... -Darcadedb.dumpMetricsEvery=10`

You can access the profile with `Profiler.INSTANCE`

Every embedded database instance collects statistics internally. You can access to them with:

```java
Map<String, Object> dbStats = db.getStats();
```

Also, ArcadeDB Server provides JMX access to the profiler (remotely). This is the class, but it's just the beginning: https://github.com/ArcadeData/arcadedb/blob/aa5229a0dc9b0aa02ce4a1da2d36497d5b6e03d8/server/src/main/java/com/arcadedb/server/JMXServerMetrics.java.
